pkgs/applications/networking/cluster/ssm-agent/default.nix

```
@@ -63,6 +63,9 @@ buildGoPackage rec {
     substituteInPlace agent/session/shell/shell_unix.go \
         --replace '"script"' '"${util-linux}/bin/script"'
 
+    substituteInPlace agent/appconfig/constants_unix.go \
+        --replace '"/etc/amazon/ssm/"' 'os.Getenv("SSM_AGENT_ETC") + "/"'
```

I think this should work, and seems better than patching in a nonstandard env var.

```suggestion
        --replace '"/etc/amazon/ssm/"' '"${placeholder "out"}/etc/amazon/ssm/"'
```

---

I went with an ENV var in case someone wanted to override the location with their own configuration file. Is there another way to preserve that functionality? I initially wanted to go the way you're suggesting

---

Is there a reason one would want to change this config?

If so, this should probably be a patch to look in `$out/etc/amazon/ssm` when `SSM_AGENT_ETC` isn't set. By wrapping with `--set`, the override wouldn't work anyways (it would always take the value of the `--set` env var, AFAIK).

---

Maybe I'm overthinking it. I'll stick with the default config. I'll make all the suggested changes tomorrow. Thanks for the feedback.

---

For what it's worth, you raise a good point. But as-is, our two changes are extremely similar